### PR TITLE
Improve handling of (multiple column) primary keys

### PR DESCRIPTION
**Pull request by bugthunk**
Adresses issue #11. 

Bumping version to `1.5.0.0` due to the fact that while it is true that the internal representation of a `PrimaryKey` isn't visible, its observable behaviour is radically different (i.e. retains order of columns as written in the argument list to `pkOnColumns`).

### CODE REVIEW

**Comment by bugthunk** [comment]
Travis fails on `cryptonite-0.24` for GHC `7.8.4`. I could look at it tomorrow.

**Comment by bugthunk** [comment]
Note that `sqlGetPrimaryKey` has been converted to a monad action. Why? Because doing sequential stuff was a lot easier in Haskell than in SQL. There is `with ordinality`, but that constrains us to `$PGVERSION >= 9.4`. There's `ROW_NUMBER() OVER()`, but I couldn't get it working satisfactorily.

**Comment by jonathanjouty** [comment]
@bugthunk 
> Travis fails on `cryptonite-0.24` for GHC `7.8.4`. I could look at it tomorrow.

Or we can just drop support for GHC < `8.*` with newer version.

**Comment by 23Skidoo** [comment]
We need to add a 8.2.2 job, BTW.

**Comment by bugthunk** [comment]
I restarted the Travis builds and all the world is green.

**Comment by 23Skidoo** [comment]
Maybe package metadata on Hackage was updated in the meantime.

File: hpqtypes-extras.cabal [hunk 1]
```diff
@@ -51,6 +51,7 @@ library
   build-depends: base < 5
                , base16-bytestring
                , bytestring
+               , Cabal
```
[the remark this answers is not in the document]
Comment (bugthunk) [reply hpqtypes-extras.cabal]: Yeah, I think I'll just do that for now.

**Comment by 23Skidoo** [comment]
Looks good to go now.

**Comment by 23Skidoo** [comment]
Once Travis is green, I'll merge this and cut a new release.

**Comment by bugthunk** [comment]
@23Skidoo ...`ghc-7.8.4` is satisfied at last :)

**Comment by 23Skidoo** [comment]
Merged, thanks!

**Comment by 23Skidoo** [comment]
New release is on Hackage.

**Pull request merged**